exoscale/api/compute.py

```diff
@@ -0,0 +1,1714 @@
+                name=name,
+                displaytext=description,
+                startip=start_ip,
+                endip=end_ip,
+                netmask=netmask,
+                networkofferingid="eb35f4e6-0ecc-412e-9925-e469bf03d8fd",  # PrivNet
+            )
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"])
+
+        return PrivateNetwork.from_cs(self, resp["network"])
+
+    def get_private_networks(self, zone=None, names=None, ids=None):
+        """
+        Get Private Networks.
+
+        Parameters:
+            zone (Zone): a zone to restrict results to
+            names ([str]): a list of Private Network names to restrict results to
+            ids ([str]): a list of Private Network identifiers to restrict results to
+
+        Returns:
+            [PrivateNetwork]: list of Private Networks
+        """
+
+        list = []
+
+        try:
+            resp = self.cs.listNetworks(
+                zoneid=getattr(zone, "id", None),
+                name=",".join(names) if names else None,
+                id=",".join(ids) if ids else None,
+            )
+
+            for i in resp.get("network", []):
+                list.append(PrivateNetwork.from_cs(self, i))
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"])
+
+        return list
+
+    def delete_private_network(self, private_network):
+        """
+        Delete a Private Network.
+
+        Parameters:
+            private_network (PrivateNetwork): the Private Network to delete
+
+        Returns:
+            None
+        """
+
+        try:
+            self.cs.deleteNetwork(id=private_network.id)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"])
+
+        return
+
+    ### Security Group
+
+    def create_security_group(self, name, description=""):
+        """
+        Create a Security Group.
+
+        Parameters:
+            name (str): the Security group name
+            description (str): the Security Group description
+
+        Returns:
+            SecurityGroup: the Security Group created
+        """
+
+        try:
+            resp = self.cs.createSecurityGroup(name=name, description=description)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"])
+
+        return SecurityGroup.from_cs(self, resp["securitygroup"])
+
+    def get_security_groups(self, name=None):
+        """
+        Get Security Groups.
+
+        Parameters:
+            name (str): a Security Group name to restrict results to
+
+        Returns:
+            [SecurityGroup]: list of Security Groups
+        """
+
+        resp = self.cs.listSecurityGroups(securitygroupname=name)
+
+        list = []
+        for i in resp.get("securitygroup", []):
+            list.append(SecurityGroup.from_cs(self, i))
+
+        return list
+
+    def delete_security_group(self, security_group):
```

Comment:
`security_group.delete()`? (it's becoming a theme isn't it? :stuck_out_tongue:)

I'll stop here but there are other places where this could apply.

Reply:
Yes, I got the message 😉  See my previous reply on this topic, and let me know if you have a strong(er) opinion than mine, in which case I'll change the implementation.